clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputDictTrainer.java

```diff
@@ -32,13 +32,15 @@
  */
 public class KafkaInputDictTrainer {
   public static class Param {
-    private String kafkaInputBroker;
-    private String topicName;
-    private String keySchema;
-    private Properties sslProperties;
-    private int compressionDictSize;
-    private int dictSampleSize;
-    private CompressionStrategy sourceVersionCompressionStrategy;
+    private final String kafkaInputBroker;
+    private final String topicName;
+    private final String keySchema;
+    private final Properties sslProperties;
+    private final int compressionDictSize;
+    private final int dictSampleSize;
+    private final CompressionStrategy sourceVersionCompressionStrategy;
+
+    private final boolean sourceVersionChunkingEnabled;
```

Comment:
Why is this change required?

Currently, this class has logic to skip the chunked messages by checking the schema id:
https://github.com/linkedin/venice/blob/main/clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputDictTrainer.java#L217

Reply:
I guess this is good to keep this change to be consistent, so that KafkaInputRecordReader will be able to execute some chunking logic, and based on the current logic, the outcome is same.

Reply:
I am actually not doing logic change in this class, but I have to populate the source version chunking info into the KafkaInputRecordReader actually. The logic here is a little bit confusing, it seems like we initiate record reader inside tranDict method (which I feel like they belong to separate logic somehow)

Reply:
Yes, `dictTrainer` needs to initialize a record read to read the messages from the source topic to build the dict for repush.